source/extensions/filters/network/redis_proxy/config.cc

```diff
@@ -49,10 +49,21 @@ Network::FilterFactoryCb RedisProxyFilterConfigFactory::createFilterFactoryFromP
 
   Upstreams upstreams;
   for (auto& cluster : unique_clusters) {
+    std::string auth_password;
+    Upstream::ThreadLocalCluster* cluster_ptr = context.clusterManager().get(cluster);
+
+    if (cluster_ptr) {
+      auto options = cluster_ptr->info()->extensionProtocolOptionsTyped<ProtocolOptionsConfigImpl>(
+          NetworkFilterNames::get().RedisProxy);
+      if (options) {
+        auth_password = options->auth_password();
+      }
+    }
+
     upstreams.emplace(cluster, std::make_shared<ConnPool::InstanceImpl>(
                                    cluster, context.clusterManager(),
                                    Common::Redis::Client::ClientFactoryImpl::instance_,
-                                   context.threadLocal(), proto_config.settings()));
+                                   context.threadLocal(), proto_config.settings(), auth_password));
```

Comment:
Curious, but could fetching the password for the upstream cluster be done in the constructor since the connection pool instance has access to everything it needs?

Reply:
yes, but I ended up with a bazel circular dependency, and decided that it would be best to keep everything in config.cc. I'll look into it more, and see if I can tease apart the problem.

Reply:
I've teased it apart, and just pushed new changes. The authentication password is now set in the ThreadLocalPool constructor.